core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/heuristic/selector/common/iterator/AbstractRandomChangeIterator.java

```diff
@@ -53,7 +57,11 @@ protected Move_ createUpcomingSelection() {
             entity = entityIterator.next();
             valueIterator = valueSelector.iterator(entity);
         }
+        Object existingValue = entityVariableDescriptor.getValue(entity);
         Object toValue = valueIterator.next();
+        if (Objects.equals(existingValue, toValue)) {
```

Comment:
Use `==` because the user domain values might override equals().

Reply:
Why would that be an issue? If they're equal, they're equal.
Not doing equals() here will ensure that no two `int`s are ever equals, because they get auto-boxed to different instances.